portfolio/src/main/java/com/google/sps/storage/DatastoreCommentStorage.java

```diff
@@ -0,0 +1,83 @@
+package com.google.sps.storage;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.sps.storage.CommentStorage;
+import com.google.sps.data.Comment;
+import java.util.ArrayList;
+import java.util.List;
+
+
+public class DatastoreCommentStorage implements CommentStorage {
+  private final DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
+  private static final String ENTITY_QUERY = "Comment";
+  private static final String ENTITY_TEXT_PARAM = "comment-text";
+  private static final String ENTITY_NAME_PARAM = "username";
+  private static final String ENTITY_TIME_PARAM = "time-posted";
+
+  @Override
+  public void insert(Comment comment) {
+    String name = comment.name();
+
+    if (comment.name().equals("")) {
+      name = "Anonymous";
+    }
+
+    Entity commentEntity = new Entity(ENTITY_QUERY);
+    commentEntity.setProperty(ENTITY_TEXT_PARAM, comment.text());
+    commentEntity.setProperty(ENTITY_NAME_PARAM, name);
+    commentEntity.setProperty(ENTITY_TIME_PARAM, comment.timePosted());
+
+    datastore.put(commentEntity);
+  }
+
+  @Override
+  public ArrayList<Comment> getNComments(int numComments, boolean sortAsc) {
+
+    Query commentQuery;
+    if (sortAsc) {
+      commentQuery = new Query("Comment").addSort(ENTITY_TIME_PARAM, SortDirection.ASCENDING);
```

Comment:
Don't inline "Comment" here. If it diverges from ENTITY_QUERY the system breaks.

This whole block could probably be:
```java
Query commentQuery = new Query(ENTITY_QUERY);
if (sortAsc) {
  commentQuery.addSort(...
...
```